internal/eblog/coder.go

```diff
@@ -0,0 +1,103 @@
+package eblog
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+	"github.com/coder/coder/v2/agent/proto"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/agentsdk"
+	"github.com/coder/retry"
+	"github.com/google/uuid"
+	"storj.io/drpc"
+)
+
+var (
+	rpcConnectTimeout  = 10 * time.Second
+	logSendGracePeriod = 10 * time.Second
+)
+
+// Coder establishes a connection to the Coder instance located at
+// coderURL and authenticates using token. It then establishes a
+// dRPC connection to the Agent API and begins sending logs.
+// The returned function is used to block until all logs are sent.
+func Coder(ctx context.Context, coderURL *url.URL, token string) (LogFunc, func(), error) {
```

Comment:
To remove the possibility of this breaking older users, I can check the Coder version and conditionally return a different implementation that uses the old deprecated endpoint. WDYT?

Reply:
We can technically break as we are bumping a major version.

Reply:
I ended up adding support for backward compatibility based on a similar commit in [envbox](https://github.com/coder/envbox/commit/4e0a070ae372cb1433176df815e69d7e0bcd8443).